portal/portal-charon/charon/src/webapp/scripts/themeswitcher.js

```diff
@@ -1,43 +1,84 @@
 function themeSwitcher(){
 
     let themeSwitcher = document.getElementById("themeSwitcher");
+    const darkThemeClass = 'sakai-dark-theme';
 
-    SAK_THEME_SELECTION_KEY = portal.user.id + "#theme";
+    if(themeSwitcher !== null) {
+        themeSwitcher.addEventListener('click', toggleDarkTheme, false);
+    }
 
-    if(themeSwitcher != null) {
-        
-        initTheme(); // on page load, if user has already selected a specific theme -> apply it
+    if(isLoggedIn()) {
+        // only check for null preference because light and dark themes are already set by Java
+        if( isPortalThemeUserPrefNull() ) {
+            if(isOsDarkThemeSet()) {
+                setThemeSwitcherToggle(true);
+                addThemeToHTMLTag(darkThemeClass);
+                setPortalThemeUserPref(true);
+            } else {
+                // to define a user preference:
+                setPortalThemeUserPref(false);
+            }
+        }
+    } else if(isOsDarkThemeSet()) {
+        addThemeToHTMLTag(darkThemeTag);
+    }
 
-        themeSwitcher.addEventListener('click', setTheme, false);
+    function addThemeToHTMLTag(themeClass) {
+        document.documentElement.classList.add(themeClass);
+    }
 
+    function removeThemeFromHTMLTag(themeClass) {
+        document.documentElement.classList.remove(themeClass);
     }
 
-    function setTheme() {
+    function isOsDarkThemeSet() {
+        if(window.matchMedia('(prefers-color-scheme: dark)').matches) {
+            return true;
+        } else {
+            return false;
+        }
+    }
 
-        // Set to dark mode
-        if (themeSwitcher.getAttribute("aria-checked") === "true") {
-            themeSwitcher.setAttribute("aria-checked", "false");
-            document.firstElementChild.classList.add('sakai-dark-theme');
-            localStorage.setItem('sakai-theme', 'dark');
+    function isLoggedIn() {
+        if (portal.user.id) {
+            return true;
+        } else {
+            return false;
         }
-        // Set to light mode
-        else {
-            themeSwitcher.setAttribute("aria-checked", "true");
-            document.firstElementChild.classList.remove('sakai-dark-theme');
-            localStorage.setItem('sakai-theme', 'light')
+    }
+
+    function isPortalThemeUserPrefNull() {
+        if (portal.themeUserPref === "-1") {
+            return true;
+        } else {
+            return false;
         }
```

Comment:
Same here:

```suggestion
        return portal.themeUserPref === "-1";
```

Reply:
Yes, this was just temporary; I'm planning to replace this.